### ISSUE TITLE
关于多分辨率加速问题

### ISSUE THREAD

**Issue by lovejing0306** [siliconflow/onediff#727]
请问当多分辨率推理时，是不支持加速吗？

我的代码如下：
```
# model load and compile
pipeline = StableDiffusionXLImg2ImgPipeline.from_pretrained(
    model_dir,
    unet=unet,
    torch_dtype=torch.float16,
    variant="fp16",
)
pipeline.scheduler = LCMScheduler.from_config(pipeline.scheduler.config)
pipeline.safety_checker = None
pipeline.to('cuda', torch_dtype=torch.float16)

pipeline.unet = oneflow_compile(pipeline.unet)
pipeline.vae.decoder = oneflow_compile(pipeline.vae.decoder)

# warm up
_ = pipeline(prompt=prompt, height=1024, width=1024)
_ = pipeline(prompt=prompt, height=2048, width=2048)
_ = pipeline(prompt=prompt, height=512, width=512)

# norm infer
for size in [(1024, 1024), (512, 2048), (2048, 512)]:
    pipeline(prompt=prompt, height=size[0], width=size[1])

```

**Comment by strint** [comment]
按说执行一次 warmup 后，各种分辨率就都可以加速了。你是发现 norm infer 没有加速么？

可以看看执行的 iter/s 的速度对比么。

**Comment by lovejing0306** [comment]
> 按说执行一次 warmup 后，各种分辨率就都可以加速了。你是发现 norm infer 没有加速么？
> 
> 可以看看执行的 iter/s 的速度对比么。

我这边是在 A100 上测试的，对单分辨率 1024x1024 是可以加速的，推理速度大概是 390ms，我这边运行的代码如下：

```
# model load and compile
pipeline = StableDiffusionXLImg2ImgPipeline.from_pretrained(
    model_dir,
    unet=unet,
    torch_dtype=torch.float16,
    variant="fp16",
)
pipeline.scheduler = LCMScheduler.from_config(pipeline.scheduler.config)
pipeline.safety_checker = None
pipeline.to('cuda', torch_dtype=torch.float16)

pipeline.unet = oneflow_compile(pipeline.unet)
pipeline.vae.decoder = oneflow_compile(pipeline.vae.decoder)

# warm up
_ = pipeline(prompt=prompt, height=1024, width=1024)

# norm infer
for size in [(1024, 1024), (1024, 1024), (1024, 1024), (1024, 1024), (1024, 1024)]:
    pipeline(prompt=prompt, height=size[0], width=size[1])

```

对于多分辨率，用的大代码就是我就开始描述问题的写法，推理速度大概是 680ms，没有起到加速的效果。

是我哪里写的不对吗？

**Comment by hjchen2** [comment]
应该是你预热的分辨率和正常推理用的分辨率不一样导致的吧，你可以尝试改一下，

```python
# warm up
_ = pipeline(prompt=prompt, height=1024, width=1024)
_ = pipeline(prompt=prompt, height=512, width=2048)
_ = pipeline(prompt=prompt, height=2048, width=512)

# norm infer
for size in [(1024, 1024), (512, 2048), (2048, 512)]:
    pipeline(prompt=prompt, height=size[0], width=size[1])

```

**Comment by lovejing0306** [comment]
> 应该是你预热的分辨率和正常推理用的分辨率不一样导致的吧，你可以尝试改一下，
> 
> ```python
> # warm up
> _ = pipeline(prompt=prompt, height=1024, width=1024)
> _ = pipeline(prompt=prompt, height=512, width=2048)
> _ = pipeline(prompt=prompt, height=2048, width=512)
> 
> # norm infer
> for size in [(1024, 1024), (512, 2048), (2048, 512)]:
>     pipeline(prompt=prompt, height=size[0], width=size[1])
> ```

试了下你说的这种方式，依然无法起大加速效果。

**Comment by lixiang007666** [comment]
我补全了你基于 sdxl 的图像到图像脚本，然后进行测试：


```python
from PIL import Image
import oneflow as flow
import torch

from onediff.infer_compiler import oneflow_compile
from diffusers import StableDiffusionXLImg2ImgPipeline, LCMScheduler
from diffusers.utils import load_image

# Model load and compile
pipeline = StableDiffusionXLImg2ImgPipeline.from_pretrained(
    "stabilityai/stable-diffusion-xl-base-1.0",
    torch_dtype=torch.float16,
    variant="fp16",
    use_safetensors=True,
)
pipeline.scheduler = LCMScheduler.from_config(pipeline.scheduler.config)
pipeline.safety_checker = None
pipeline.to('cuda', torch_dtype=torch.float16)

pipeline.unet = oneflow_compile(pipeline.unet)
pipeline.vae.decoder = oneflow_compile(pipeline.vae.decoder)

url = "https://huggingface.co/datasets/patrickvonplaten/images/resolve/main/aa_xl/000000009.png"
init_image = load_image(url).convert("RGB")
prompt = "a photo of an astronaut riding a horse on mars"

def resize_image(image, size):
    return image.resize(size, Image.LANCZOS)

# Warm-up
warmup_sizes = [(1024, 1024)]
for size in warmup_sizes:
    resized_image = resize_image(init_image, size)
    _ = pipeline(prompt=prompt, image=resized_image)

# Normal inference
inference_sizes = [(1024, 1024), (512, 2048), (2048, 512)]
for size in inference_sizes:
    resized_image = resize_image(init_image, size)
    image = pipeline(prompt=prompt, image=resized_image).images[0]

```

我们支持动态切换分辨率加速，输出如下：

![image](https://github.com/siliconflow/onediff/assets/54010254/73efcd24-b2a5-4239-9390-57b68a4be13c)


当不使用 onediff 加速，也就是注释掉：
```
pipeline.unet = oneflow_compile(pipeline.unet)
pipeline.vae.decoder = oneflow_compile(pipeline.vae.decoder)
```
输出如下：

![image](https://github.com/siliconflow/onediff/assets/54010254/64252d32-92b0-4562-99c3-454a2eff105b)

**Comment by strint** [comment]
@lovejing0306 我们没有复现你的问题。可以参考上面的例子再试下

**Comment by lovejing0306** [comment]
> 我补全了你基于 sdxl 的图像到图像脚本，然后进行测试：
> 
> ```python
> from PIL import Image
> import oneflow as flow
> import torch
> 
> from onediff.infer_compiler import oneflow_compile
> from diffusers import StableDiffusionXLImg2ImgPipeline, LCMScheduler
> from diffusers.utils import load_image
> 
> # Model load and compile
> pipeline = StableDiffusionXLImg2ImgPipeline.from_pretrained(
>     "stabilityai/stable-diffusion-xl-base-1.0",
>     torch_dtype=torch.float16,
>     variant="fp16",
>     use_safetensors=True,
> )
> pipeline.scheduler = LCMScheduler.from_config(pipeline.scheduler.config)
> pipeline.safety_checker = None
> pipeline.to('cuda', torch_dtype=torch.float16)
> 
> pipeline.unet = oneflow_compile(pipeline.unet)
> pipeline.vae.decoder = oneflow_compile(pipeline.vae.decoder)
> 
> url = "https://huggingface.co/datasets/patrickvonplaten/images/resolve/main/aa_xl/000000009.png"
> init_image = load_image(url).convert("RGB")
> prompt = "a photo of an astronaut riding a horse on mars"
> 
> def resize_image(image, size):
>     return image.resize(size, Image.LANCZOS)
> 
> # Warm-up
> warmup_sizes = [(1024, 1024)]
> for size in warmup_sizes:
>     resized_image = resize_image(init_image, size)
>     _ = pipeline(prompt=prompt, image=resized_image)
> 
> # Normal inference
> inference_sizes = [(1024, 1024), (512, 2048), (2048, 512)]
> for size in inference_sizes:
>     resized_image = resize_image(init_image, size)
>     image = pipeline(prompt=prompt, image=resized_image).images[0]
> ```
> 
> 我们支持动态切换分辨率加速，输出如下：
> 
> ![image](https://private-user-images.githubusercontent.com/54010254/313686358-73efcd24-b2a5-4239-9390-57b68a4be13c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA4MzcxMjMsIm5iZiI6MTcxMDgzNjgyMywicGF0aCI6Ii81NDAxMDI1NC8zMTM2ODYzNTgtNzNlZmNkMjQtYjJhNS00MjM5LTkzOTAtNTdiNjhhNGJlMTNjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE5VDA4MjcwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY0YTM1Y2Y0ODg0ODcyZDJiMTcyZjk0NjY4MzAyNGRkNjAwNWQ4OWFhNTMyZTgyMzUxOTBmOGI1ZDBhZmM0ZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ggrCkWQTWXuIK547euvm6IdaqPo9udGcorjkMeukveY)
> 
> 当不使用 onediff 加速，也就是注释掉
> 
> ```
> pipeline.unet = oneflow_compile(pipeline.unet)
> pipeline.vae.decoder = oneflow_compile(pipeline.vae.decoder)
> ```
> 
> ，输出如下：
> 
> ![image](https://private-user-images.githubusercontent.com/54010254/313686464-64252d32-92b0-4562-99c3-454a2eff105b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA4MzcxMjMsIm5iZiI6MTcxMDgzNjgyMywicGF0aCI6Ii81NDAxMDI1NC8zMTM2ODY0NjQtNjQyNTJkMzItOTJiMC00NTYyLTk5YzMtNDU0YTJlZmYxMDViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE5VDA4MjcwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZlYjU0MGQyNDBhM2E1ZjgxYTMzMjk2NTA0OWY1NDg2YWRjM2I1YWUxZDUxMzcyOWFjMjAwNzgzZjFjOWMyOGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Ql-8Ut5VSan7_NjGf_B6o4ZSZLKwWuCGnMLWRCyv44o)
> 
> 按照我的代码，check 你的脚本。
> 
> **补充**：你的 diffusers 版本是什么？

用你的这种写法可以实现了，多谢！！！
